### ISSUE TITLE
Cannot join servers that have WorldEdit installed?

### ISSUE THREAD

**Issue by ReduxRedstone** [PrismarineJS/node-minecraft-protocol#444]
For some reason if a server has the World Edit plugin installed this lib fails to connect correctly. I was beating my self over the head for hours trying to figure out why it was not connecting. After removing all the plugins and adding them back one-by-one I narrowed the issue down to the World Edit plugin. for whatever reason it kicks the client after one tick and I am unsure how to fix this. No errors are logged in the command prompt, it just stops the script. In the server console it reads:
`lost connection: Internal Exception: java.io.IOException: Error while read(...): Connection reset by peer`
It gets even stranger when I try to join the server using a chat client I wrote which uses socket.io. If I use the chat client, then it doesn't kick the client after one tick, in fact it allows it stay on for a few seconds before giving a `Timed out` error. The client is able to _send_ packets during this time, but cannot _read_ packets (I was successful in sending chat messages, by checking console, however the client did not seem to be able to read the packets, and did not echo them back like it usually does).

I'm still fairly new to this lib, so it could very well just be something I'm doing wrong, but the issue even occurs when using ONLY this code:
```javascript
const mc = require('minecraft-protocol');

var client = mc.createClient({
  username: "email",
  password: "password",
  host: "ip",
  port: 25565,
  version: "1.10.2"
});
```
Which is the absolute bare minimum, so I do not think it's something I am doing.

Before you suggest it, no the host itself is not blocking me in any way. I downloaded a local copy of the server and tested it using localhost and the same issue persisted.

Any ideas or fixes? I would have liked to release my chat client, but this error really limits the number of servers it will work on.

**Comment by rom1504** [comment]
It seems world edit uses forge (https://github.com/sk89q/WorldEdit) so it might be required to enable forge support, see https://github.com/PrismarineJS/node-minecraft-protocol-forge

**Comment by ReduxRedstone** [comment]
WorldEdit the mod uses Forge, but WorldEdit the plugin does not. I am able to join servers that have world edit the plugin (not mod) installed perfectly fine in a vanilla client without forge, so I don't see why I would need to use the forge protocol when trying it with this lib. Just in case though, I installed the forge protocol and changed the code to:
```javascript
const mc = require('minecraft-protocol');
const autoVersionForge = require('minecraft-protocol-forge').autoVersionForge;

var client = mc.createClient({
  username: "halolink44@gmail.com",
  password: "linkman444",
  host: "play.citadelrpg.net",
  port: 25565,
  version: "1.10.2"
});
autoVersionForge(client);
client.on('error', function(packet) {
  console.log(packet);
});
```
as shown on the GitHub you linked, but the same error still persists. Changing the version to `false` instead of `1.10.2` wont connect me to the server at all.

**Comment by Gjum** [comment]
is that your real password?

**Comment by ReduxRedstone** [comment]
I resolved this issue by using this fork of the protocol
https://github.com/roblabla/node-minecraft-protocol/tree/8c02735af1fbe2b9b98e56824f838871ab2eb678
which now allows me to join the server perfectly fine. I do have one other question, though, which is extremely trivial, it probably doesnt need its own post. When ever I do `client.end()` it doesnt seem to fully log the client out. Whenever I connect to a server, then do `client.end()`, and then join another server without closing the Node process, it always gives me an "Invalid Token" error. I need to close the whole Node process in order for it to fully logout, which isnt very desirable for the project im working on (a chat client that lets you log in and out of different servers, like Minechat but in browser. It all works except the "Invalid Token" error I keep getting when changing servers)

**Comment by roblabla** [comment]
Welp. I should really merge #434. We'll fix the code to make it prettier later. What do you think @rom1504 ?

**Comment by rom1504** [comment]
Yeah let's merge it

**Comment by rom1504** [comment]
now fixed in 1.1.1
